pkg/controller/sparkapplication/spark_pod_eventhandler.go

```diff
@@ -68,13 +74,24 @@ func (s *sparkPodEventHandler) onPodDeleted(obj interface{}) {
 	if deletedPod == nil {
 		return
 	}
-	glog.V(2).Infof("Pod %s deleted in namespace %s.", deletedPod.GetObjectMeta().GetName(), deletedPod.GetObjectMeta().GetNamespace())
+	glog.V(2).Infof("Pod %s deleted in namespace %s.", deletedPod.GetName(), deletedPod.GetNamespace())
 	s.enqueueSparkAppForUpdate(deletedPod)
 }
 
 func (s *sparkPodEventHandler) enqueueSparkAppForUpdate(pod *apiv1.Pod) {
-	if appKey, ok := createMetaNamespaceKey(pod); ok {
-		glog.V(2).Infof("Enqueuing SparkApplication %s for app update processing.", appKey)
-		s.enqueueFunc(appKey)
+	appName, exists := getAppName(pod)
+	if !exists {
+		return
 	}
+
+	if submissionID, exists := pod.Labels[config.SubmissionIDLabel]; exists {
+		app, err := s.applicationLister.SparkApplications(pod.GetNamespace()).Get(appName)
+		if err != nil || app.Status.SubmissionID != submissionID {
```

Comment:
Recently we run into an issue where pods stuck in terminating status (several people have reported this on K8s project), the question is whether the operator should be responsible for cleaning stuff in the background. For now we can ignore this and keep submitting.

Reply:
I think essentially we want to wait for the driver pod to be completely gone. This is actually handled in `validateSparkResourceDeletion` under the `PendingRerun` state. If `validateSparkResourceDeletion` returns false, the state will remain `PendingRerun` and the next resync loop will check again.

Reply:
That means `validateSparkResourceDeletion` will always return a stuck pod. Should we add a configurable policy to handle this, like retry a force delete after N time passes?

Reply:
While, a pod has a `terminationGracePeriod` after which the pod will be sent a SIGKILL. So once deletion of the pod is requested, the pod should be gone after certain amount of time.